Software/src/main/java/org/foi/emp/carmanagement/controllers/UserController.java

```diff
@@ -45,4 +53,19 @@
             return new ResponseEntity<>("User doesn t exist in database and can t been deleted!", HttpStatus.NOT_FOUND);
         }
     }
+
+    @GetMapping("/{userId}/{activationCode}")
+    public ResponseEntity<String> activateUser(@PathVariable("userId") Long id, @PathVariable("activationCode") String activationCode) {
+        Optional<UserModel> userModel = this.userServices.getUserById(id);
+        if (userModel.isPresent() && StringUtils.equals(activationCode, userModel.get().getActivationCode())) {
+            UserModel model=userModel.get();
```

Comment:
Replace this `UserModel model=userModel.get();` with this `final UserModel model=userModel.get();`